Makefile

```
@@ -0,0 +1,504 @@
+	release-build
+	mkdir -p crypto-ffi/bindings/Swift/WireCoreCryptoUniffi/WireCoreCryptoUniffi
+	$(UNIFFI_BINDGEN) generate \
+	  --language swift \
+	  --out-dir crypto-ffi/bindings/Swift/WireCoreCryptoUniffi/WireCoreCryptoUniffi \
+	  --library target/release/libcore_crypto_ffi.$(LIBRARY_EXTENSION)
+	$(TOUCH_STAMP)
+endif
+
+.PHONY: bindings-swift swift
+bindings-swift: $(STAMPS)/bindings-swift ## Generate Swift bindings
+swift: $(STAMPS)/bindings-swift $(STAMPS)/docs-swift
+
+# Kotlin-Android bindings
+$(STAMPS)/bindings-kotlin-android: target/release/uniffi-bindgen release-build
+	mkdir -p crypto-ffi/bindings/android/src/main/uniffi
+	$(UNIFFI_BINDGEN) generate \
+	  --config uniffi-android.toml \
+	  --language kotlin \
+	  --no-format \
+	  --out-dir crypto-ffi/bindings/android/src/main/uniffi \
+	  --library target/release/libcore_crypto_ffi.$(LIBRARY_EXTENSION)
+	$(TOUCH_STAMP)
+
+.PHONY: bindings-kotlin-android
+bindings-kotlin-android: $(STAMPS)/bindings-kotlin-android ## Generate Kotlin bindings for Android
+
+# Kotlin-JVM bindings
+$(STAMPS)/bindings-kotlin-jvm: target/release/uniffi-bindgen release-build
+	mkdir -p crypto-ffi/bindings/jvm/src/main/uniffi
+	$(UNIFFI_BINDGEN) generate \
+	  --language kotlin \
+	  --no-format \
+	  --out-dir crypto-ffi/bindings/jvm/src/main/uniffi \
+	  --library target/release/libcore_crypto_ffi.$(LIBRARY_EXTENSION)
+	$(TOUCH_STAMP)
+
+.PHONY: bindings-kotlin-jvm
+bindings-kotlin-jvm: $(STAMPS)/bindings-kotlin-jvm ## Generate Kotlin bindings for JVM
+
+# Grouped Kotlin bindings
+.PHONY: bindings-kotlin
+bindings-kotlin: bindings-kotlin-android bindings-kotlin-jvm ## Generate all Kotlin bindings
+
+#-------------------------------------------------------------------------------
+# 4) WASM build + JS deps via stamps
+#-------------------------------------------------------------------------------
+
+# The default build condition is `dev`, which is much faster.
+# To enable optimizations, set the `WASM_RELEASE` variable to any non-empty value:
+#
+# i.e.
+#
+#   make wasm WASM_RELEASE=1
+WASM_BUILD_ARGS := $(if $(WASM_RELEASE),,--dev)
+
+# Generate WASM package
+$(STAMPS)/wasm-build: FORCE
+	cd crypto-ffi && \
+	$(WASM_PACK) build \
+	  --locked \
+	  --no-pack \
+	  --out-dir bindings/js/src/autogenerated \
+	  --out-name core-crypto-ffi \
+	  --mode normal \
+	  --target web \
+	  $(WASM_BUILD_ARGS)
+	$(TOUCH_STAMP)
+
+.PHONY: wasm-build
+wasm-build: $(STAMPS)/wasm-build
+
+# Install JS deps with Bun
+$(STAMPS)/bun-deps: $(PACKAGE_JSON)
+	$(BUN) install --frozen-lockfile --cwd crypto-ffi/bindings/js
+	$(TOUCH_STAMP)
+
+.PHONY: bun-deps
+bun-deps: $(STAMPS)/bun-deps ## Install JS dependencies using bun
+
+# Full JS binding step
+bindings-js: wasm-build $(JS_OUT) ## Generate JavaScript bindings (append `WASM_RELEASE=1` to this command or any dependents for the release build)
+
+# All bindings
+.PHONY: bindings
+bindings: bindings-swift bindings-kotlin bindings-js ## Generate all bindings
```

```suggestion
bindings: bindings-kotlin bindings-js $(if $(filter Darwin,$(UNAME_S)),bindings-swift) ## Generate all bindings
```

No need to generate a warning line about Swift every time we generate bindings locally.